ir/memory.h

```diff
@@ -287,6 +287,9 @@ class Memory {
 
   unsigned num_nonlocals() const;
 
+  const auto &get_local_block_val() const { return local_block_val; }
```

Comment:
these can't be exposed. These are internal details of the memory model

Reply:
They are used at strlen() - do you prefer moving its encoding into Memory class (as memcpy/memset does)?

Reply:
Not sure strlen() needs to go to memory.cpp. The exported load() is not sufficient?

Reply:
Nope, it isn't sufficient because the strlen recursive function needs to have arguments for the current memory.
The stored values at local_block_val and non_local_block_val are quantified, and they can be used by recursive functions by passing through arguments only, I think.

Reply:
Or we can declare Strlen as a friend class of Memory and hide these.

Reply:
The point is that we really don't want to expose the internals of the memory model.
Instead of quantifying the memory, you can use them directly and create one strlen() function for each call.
We can memoize them by having the memory produce a unique string that deterministically hashes the state. This is easy (just return the Z3 ids of the 2 block val arrays, for example). This will ensure that source/target blocks that are equivalent keep their formulas syntactically equal.